js/client-ladder.js

```diff
@@ -135,22 +136,28 @@
 				var self = this;
 				this.$el.html('<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><em>Loading...</em></p></div>');
 				if (app.localLadder) {
-					app.send('/cmd laddertop ' + format);
+					app.send('/cmd laddertop ' + format + (this.curSearchVal ? ' ,' + this.curSearchVal : ''));
 				} else {
 					$.get('/ladder.php', {
 						format: format,
 						server: Config.server.id.split(':')[0],
-						output: 'html'
+						output: 'html',
+						prefix: this.curSearchVal
 					}, function (data) {
 						if (self.curFormat !== format) return;
-						var buf = '<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><button class="button" name="refresh"><i class="fa fa-refresh"></i> Refresh</button></p>';
-						buf += '<h3>' + BattleLog.escapeFormat(format) + ' Top 500</h3>';
+						var buf = '<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><button class="button" name="refresh"><i class="fa fa-refresh"></i> Refresh</button>';
+						buf += '<form><input type="text" id="ladderSearchBar" name="search" class="textbox searchinput" value="' + (self.curSearchVal || '') + '" placeholder="username prefix"/><button name="search" type="submit"> Search</button></form></p>';
```

Comment:
```suggestion
						buf += '<form><input type="text" id="ladderSearchBar" name="search" class="textbox searchinput" value="' + (self.curSearchVal || '') + '" placeholder="username prefix"/><button type="submit"> Search</button></form></p>';
```

Reply:
Oh, wait, I understand why your submit button is named `search` now.

Reply:
Yeah, its kind of weird that theyre both named `search`? Suggestions?

Reply:
```suggestion
						buf += '<form class="search"><input type="text" name="searchval" class="textbox searchinput" value="' + (self.curSearchVal || '') + '" placeholder="username prefix"/><button type="submit"> Search</button></form></p>';
```

- Form submission should rely on the `submit` event. Relying on the `click` event of the submit button is unreliable.
- You should not have both an `<input>` and a `<button>` named `search` – names within a single form should be unique.
- You should not use the `id` attribute, ever (with rare exceptions). It's HTML/CSS's equivalent of a global variable. If you want to make an element findable, `class` is the usual alternative, and specifically `name` is the alternative for form properties.